apps/web/src/components/Composer/NewPublication.tsx

```diff
@@ -907,39 +972,90 @@ const NewPublication: FC<NewPublicationProps> = ({ publication }) => {
-            <>
-              <CollectSettings />
-              <ReferenceSettings />
-              <AccessSettings />
-            </>
-          )}
-          <PollSettings />
-        </div>
-        <div className="ml-auto pt-2 sm:pt-0">
-          <Button
-            disabled={
-              isLoading ||
-              isUploading ||
-              isSubmitDisabledByPoll ||
-              videoThumbnail.uploading
-            }
-            icon={
-              isLoading ? (
-                <Spinner size="xs" />
-              ) : isComment ? (
-                <ChatAlt2Icon className="h-4 w-4" />
-              ) : (
-                <PencilAltIcon className="h-4 w-4" />
-              )
-            }
-            onClick={createPublication}
-          >
-            {isComment ? t`Comment` : t`Post`}
-          </Button>
+        {showNewPublicationModal &&
+        modalPublicationType === NewPublicationTypes.Spaces ? (
+          <SpaceSettings />
+        ) : (
+          <div className="flex items-center space-x-4">
+            <Attachment />
+            <Gif setGifAttachment={(gif: IGif) => setGifAttachment(gif)} />
+            {!publication?.isDataAvailability && (
+              <>
+                <CollectSettings />
+                <ReferenceSettings />
+                <AccessSettings />
+              </>
+            )}
+            <PollSettings />
+          </div>
+        )}
+        <div className="absolute bottom-2 right-2 inline-flex">
+          <div className="ml-auto pt-2 sm:pt-0">
+            <Button
+              disabled={
+                isLoading ||
+                isUploading ||
+                isSubmitDisabledByPoll ||
+                videoThumbnail.uploading
+              }
+              icon={
+                isLoading ? (
+                  <Spinner size="xs" />
+                ) : isComment ? (
+                  <ChatAlt2Icon className="h-4 w-4" />
+                ) : showNewPublicationModal &&
+                  modalPublicationType === NewPublicationTypes.Spaces ? (
+                  <MicrophoneIcon className="h-4 w-4" />
+                ) : (
+                  <PencilAltIcon className="h-4 w-4" />
+                )
+              }
+              onClick={createPublication}
+            >
+              {isComment
+                ? t`Comment`
+                : showNewPublicationModal &&
+                  modalPublicationType === NewPublicationTypes.Spaces
+                ? t`Create spaces`
+                : t`Post`}
+            </Button>
+          </div>
+          {showNewPublicationModal &&
+            modalPublicationType === NewPublicationTypes.Spaces && (
+              <Dropdown
+                triggerChild={
+                  <div className="ml-2 inline-flex h-8 w-8 items-center justify-center rounded-md border border-violet-500 p-1">
+                    <div className="text-brand-500 relative h-6 w-6">
+                      {Icons.calendar}
```

Comment:
If we render icons like this wouldn't it import all the icons in the object altogether no matter it is used or not right? @bigint

Reply:
Icons IMO it should be from heroicons, it has all we need @vrajdesai78

Reply:
I have updated most of the icons with heroicons, but I am unable to find 3 icons, which I have still kept in Icons.tsx.